frontend/src/js/external-forms/form-query-node-editor/FormQueryNodeEditor.tsx

```diff
@@ -58,51 +54,21 @@ interface PropsT {
 }
 
 const FormQueryNodeEditor = (props: PropsT) => {
-  const [editedNode, setEditedNode] = useState(props.node);
-
-  const previousNodePosition = usePrevious(props.nodePosition);
-  useEffect(
-    function () {
-      if (previousNodePosition !== props.nodePosition) {
-        setEditedNode(
-          initTables({
-            blocklistedTables: props.blocklistedTables,
-            allowlistedTables: props.allowlistedTables,
-          })(props.node),
-        );
-      }
-    },
-    [
-      previousNodePosition,
-      props.nodePosition,
-      props.node,
-      props.blocklistedTables,
-      props.allowlistedTables,
-    ],
-  );
-
-  useEffect(
-    function syncWithNodeFromOutside() {
-      setEditedNode(props.node);
-    },
-    [props.node],
-  );
```

Comment:
This + most of the other changes simplify future debugging of the update logic within FormQueryNodeEditor